applications/dashboard/settings/structure.php

```diff
@@ -851,6 +851,19 @@
     ->column('UpdateIPAddress', 'ipaddress', true)
     ->set($Explicit, $Drop);
 
+$Construct
+    ->table("contentDraft")
+    ->primaryKey("draftID")
+    ->column("recordType", "varchar(64)", false, ["index", "index.record", "index.parentRecord"])
+    ->column("recordID", "int", true, "index.record")
+    ->column("parentRecordID", "int", true, "index.parentRecord")
+    ->column("attributes", "text")
+    ->column("insertUserID", "int", false, "index")
+    ->column("dateInserted", "datetime")
+    ->column("updateUserID", "int")
+    ->column("dateUpdated", "datetime")
+    ->set($Explicit, $Drop);
+
```

Comment:
So:

* I understand our API-first methodology expects camelCase parameter names.
* I understand we're currently shimming every API endpoint to transform our CapitalCase database structure into a camelCase schema.

But:

* We have no plan or priority for actually migrating our database to the new standard.
* There is little-to-no relevant performance overhead incurred by the shimming.
* There is little technical debt to putting the shim in new endpoints. (They can just be deleted when we move with the rest of them.)
* There are no extensive plans for additional core tables beyond this one presently.

Those are the technical facts on the ground. Now, enter opinion:

* You think it's more efficient to move to a new standard as soon as it's settled to minimize future code changes.
* I think deviating from established norms has real communication & training costs, both internally and in the open source community. I think those costs outweigh the benefits of halfway messing with database capitalization, especially when we have no plans to update our developer docs to reflect these changes or their rationale.

This change isn't free, and I'm not keen to absorb the cost. So the toll is gonna be a developer doc explaining our database naming conventions & why they're migrating in the public dev docs.

Reply:
For the purposes of this sprint, I will accept an IOU in the form of a docs repo issue, provided it gets in before end of year.

Reply:
Vanilla's code is *supposed* to be camel-cased. The odd man out is currently our database, because changing that is a tremendous endeavor. Not only would the database need to be updated, but every piece of code that referenced these fields would also need to be updated to avoid using the capital-case names. Column names aren't case-sensitive in MySQL, but referencing them in PHP would be.

I don't agree that we should continue perpetuating that standard, purely for the sake of consistency. It requires us to continue using mixed casing schemes in our code, even for new components. The API is a good example, where we're having to adjust casing to present the desired output, but this applies to just about anything that uses these tables. If we know this is the direction we want to go in, and apply it now, then it's one less thing to have to deal with when we make the switch. Again, this includes API endpoints, as well as any models, controllers, etc. that reference these fields with the legacy capital-case format.

Reply:
PAY THE TAX